lib/ddtrace/span/external_resource.rb

```diff
@@ -0,0 +1,38 @@
+require 'ddtrace/ext/app_types'
+require 'ddtrace/ext/http'
+
+module Datadog
+  class Span
+    # Extension for Datadog::Span that tracks if this span
+    # represents a unit of work performed by this application
+    # or an external resource being indirectly measured
+    # (e.g. SQL, Redis, outbound HTTP request).
+    module ExternalResource
```

Comment:
I would consider renaming this from `ExternalResource` to `ExternalService` or something else. The word `resource` is a term with special meaning in Datadog vernacular, and this might confuse users about it's actual intent. I think what's actually "external" here is a service, not a resource.